crates/tracing/src/layers.rs

```diff
@@ -0,0 +1,172 @@
+    } else {
+        EnvFilter::builder().from_env_lossy()
+    };
+
+    DEFAULT_ENV_FILTER_DIRECTIVES
+        .into_iter()
+        .chain(directives.split(','))
+        .try_fold(env_filter, |env_filter, directive| {
+            Ok(env_filter.add_directive(directive.parse()?))
+        })
+}
+
+/// Manages the collection of layers for a tracing subscriber.
+///
+/// `Layers` acts as a container for different logging layers such as stdout, file, or journald.
+/// Each layer can be configured separately and then combined into a tracing subscriber.
+pub(crate) struct Layers {
+    inner: Vec<BoxedLayer<Registry>>,
+}
+
+impl Layers {
+    /// Creates a new `Layers` instance.
+    pub(crate) fn new() -> Self {
+        Self { inner: vec![] }
+    }
+
+    /// Consumes the `Layers` instance, returning the inner vector of layers.
+    pub(crate) fn into_inner(self) -> Vec<BoxedLayer<Registry>> {
+        self.inner
+    }
+
+    /// Adds a journald layer to the layers collection.
+    ///
+    /// # Arguments
+    /// * `filter` - A string containing additional filter directives for this layer.
+    ///
+    /// # Returns
+    /// An `eyre::Result<()>` indicating the success or failure of the operation.
+    pub(crate) fn journald(&mut self, filter: &str) -> eyre::Result<()> {
+        let journald_filter = build_env_filter(None, filter)?;
+        let layer = tracing_journald::layer()?.with_filter(journald_filter).boxed();
+        self.inner.push(layer);
+        Ok(())
+    }
+
+    /// Adds a stdout layer with specified formatting and filtering.
+    ///
+    /// # Type Parameters
+    /// * `S` - The type of subscriber that will use these layers.
+    ///
+    /// # Arguments
+    /// * `format` - The log message format.
+    /// * `directive` - Directive for the default logging level.
+    /// * `filter` - Additional filter directives as a string.
+    /// * `color` - Optional color configuration for the log messages.
+    ///
+    /// # Returns
+    /// An `eyre::Result<()>` indicating the success or failure of the operation.
+    pub(crate) fn stdout(
+        &mut self,
+        format: LogFormat,
+        directive: Directive,
+        filter: &str,
+        color: Option<String>,
+    ) -> eyre::Result<()> {
+        let filter = build_env_filter(Some(directive), filter)?;
+        let layer = format.apply(filter, color, None);
+        self.inner.push(layer.boxed());
+        Ok(())
+    }
+
+    /// Adds a file logging layer to the layers collection.
+    ///
+    /// # Arguments
+    /// * `format` - The format for log messages.
+    /// * `filter` - Additional filter directives as a string.
+    /// * `file_info` - Information about the log file including path and rotation strategy.
+    ///
+    /// # Returns
+    /// An `eyre::Result<FileWorkerGuard>` representing the file logging worker.
+    pub(crate) fn file(
+        &mut self,
+        format: LogFormat,
+        filter: String,
```

Comment:
just for the consistency with other methods
```suggestion
        filter: &str,
```